Index: CriminalIntent/app/src/main/java/com/example/criminallintent/CrimeListFragment.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- CriminalIntent/app/src/main/java/com/example/criminallintent/CrimeListFragment.java	(revision 7ab2a66c8e309c01ec5b8863949813dff5eca6fe)
+++ CriminalIntent/app/src/main/java/com/example/criminallintent/CrimeListFragment.java	(date 1610801911057)
@@ -1,5 +1,6 @@
 package com.example.criminallintent;
 
+import android.content.Intent;
 import android.os.Bundle;
 import android.util.Log;
 import android.view.LayoutInflater;
@@ -105,9 +106,8 @@
 
         @Override
         public void onClick(View view){
-            Toast.makeText(getActivity(),
-                    mCrime.getTitle() + "clicked!", Toast.LENGTH_SHORT)
-                    .show();
+            Intent intent = MainActivity.newIntent(getActivity(), mCrime.getId());
+            startActivity(intent);
         }
     }
 }
Index: CriminalIntent/app/src/main/java/com/example/criminallintent/CrimeLab.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- CriminalIntent/app/src/main/java/com/example/criminallintent/CrimeLab.java	(revision 7ab2a66c8e309c01ec5b8863949813dff5eca6fe)
+++ CriminalIntent/app/src/main/java/com/example/criminallintent/CrimeLab.java	(date 1610802566382)
@@ -37,7 +37,7 @@
     }
 
     /*返回带指定ID的Crime对象*/
-    public Crime getCrime(UUID id){//UUID要引入java.util.UUID
+    public Crime getCrime(UUID id){
         for (Crime crime : mCrimes){
             if (crime.getId().equals(id)){
                 return crime;
Index: CriminalIntent/app/src/main/java/com/example/criminallintent/CrimeFragment.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- CriminalIntent/app/src/main/java/com/example/criminallintent/CrimeFragment.java	(revision 7ab2a66c8e309c01ec5b8863949813dff5eca6fe)
+++ CriminalIntent/app/src/main/java/com/example/criminallintent/CrimeFragment.java	(date 1610802755488)
@@ -16,6 +16,8 @@
 import androidx.fragment.app.Fragment;
 import androidx.fragment.app.FragmentManager;
 
+import java.util.UUID;
+
 public class CrimeFragment extends Fragment{
     private Crime mCrime;
     private EditText mTitleField;
@@ -25,7 +27,10 @@
     @Override
     public void onCreate(Bundle saveInstanceState){//也具有保存及获取状态的bundle
         super.onCreate(saveInstanceState);//覆盖方法的原因：Fragment生命周期方法必须是公共方法，因为托管fragment的activity要调用
-        mCrime = new Crime();
+    /*取得extra数据并取得crime对象*/
+        UUID crimeId = (UUID) getActivity().getIntent()
+                .getSerializableExtra(MainActivity.EXTRA_CRIME_ID);
+        mCrime = CrimeLab.get(getActivity()).getCrime(crimeId);
     }
 
     /*创建和配置fragment视图是另一个Fragment生命周期方法完成的*/
@@ -49,6 +54,7 @@
         mDateButton.setEnabled(false);
 
         mTitleField = (EditText) v.findViewById(R.id.crime_title);
+        mTitleField.setText(mCrime.getTitle());
         mTitleField.addTextChangedListener(new TextWatcher() {//创建实现TextWatcher监听器接口的匿名内部类？
             @Override
             public void beforeTextChanged(CharSequence s, int start, int count, int after) {
@@ -67,6 +73,7 @@
         });
 
         mSolvedCheckBox = (CheckBox) v.findViewById(R.id.crime_solved);
+        mSolvedCheckBox.setChecked(mCrime.isSolved());
         mSolvedCheckBox.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {//compound混合物，复合物；为什么就要选这种？
             @Override
             public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
Index: CriminalIntent/app/src/main/java/com/example/criminallintent/MainActivity.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- CriminalIntent/app/src/main/java/com/example/criminallintent/MainActivity.java	(revision 7ab2a66c8e309c01ec5b8863949813dff5eca6fe)
+++ CriminalIntent/app/src/main/java/com/example/criminallintent/MainActivity.java	(date 1610801575131)
@@ -3,11 +3,25 @@
 import androidx.appcompat.app.AppCompatActivity;
 import androidx.fragment.app.Fragment;
 
+import android.content.Context;
+import android.content.Intent;
 import android.os.Bundle;
 
+import java.util.UUID;
+
 public class MainActivity extends SingleFragmentActivity{
     /*抽象方法具体化,若有返回值，则返回类型要相同*/
     protected Fragment createFragment() {
         return new CrimeFragment();
     }
+    /*putExtra()所需的键名*/
+    public static final String EXTRA_CRIME_ID =
+            "com.example.criminalIntent.crime_id";
+
+    public static Intent newIntent(Context packageContext, UUID crimeId){
+        Intent intent = new Intent(packageContext, MainActivity.class);
+        intent.putExtra(EXTRA_CRIME_ID, crimeId);
+        return intent;
+    }
+
 }
\ No newline at end of file
